src/cases/components/case-hearings-list/case-hearings-list.component.ts

```diff
@@ -63,41 +70,81 @@ export class CaseHearingsListComponent implements OnInit {
+  }
+
   public viewAndEdit(hearingID: string): void {
     this.hearingStore.dispatch(new fromHearingStore.LoadHearingValues(this.caseId));
     this.hearingStore.dispatch(new fromHearingStore.LoadHearingRequest(hearingID));
-    const hearingCondition: HearingConditions = {
-      mode: Mode.VIEW,
-    };
-    this.hearingStore.dispatch(new fromHearingStore.SaveHearingConditions(hearingCondition));
-    this.router.navigate(['/', 'hearings', 'request', 'hearing-view-edit-summary']);
+    this.sub = combineLatest([this.hearingStore.select(fromHearingStore.getHearingValuesLastError),
+      this.hearingStore.select(fromHearingStore.getHearingRequestLastError)]).subscribe(
+      errors => {
+        if (errors && (errors[0] !== null || errors[1] !== null)) {
+          this.router.navigate(['/', 'hearings', 'error']);
+        } else {
+          const hearingCondition: HearingConditions = {
+            mode: Mode.VIEW,
+          };
+          this.hearingStore.dispatch(new fromHearingStore.SaveHearingConditions(hearingCondition));
+          this.router.navigate(['/', 'hearings', 'request', 'hearing-view-edit-summary']);
+        }
+      }
+    );
   }
 
   public viewDetails(hearing: HearingListViewModel): void {
-    switch (hearing.exuiDisplayStatus) {
-      case EXUIDisplayStatusEnum.CANCELLATION_REQUESTED:
-        this.hearingStore.dispatch(new fromHearingStore.LoadHearingRequest(hearing.hearingID));
-        this.router.navigate(['/', 'hearings', 'view', 'hearing-cancellation-summary']);
-        break;
-      case EXUIDisplayStatusEnum.CANCELLED:
-        this.hearingStore.dispatch(new fromHearingStore.LoadHearingRequest(hearing.hearingID));
-        this.router.navigate(['/', 'hearings', 'view', 'hearing-cancelled-summary']);
-        break;
-      case EXUIDisplayStatusEnum.COMPLETED:
-        this.hearingStore.dispatch(new fromHearingStore.LoadHearingRequest(hearing.hearingID));
-        this.router.navigate(['/', 'hearings', 'view', 'hearing-completed-summary', hearing.hearingID]);
-        break;
-      case EXUIDisplayStatusEnum.ADJOURNED:
-        this.hearingStore.dispatch(new fromHearingStore.LoadHearingRequest(hearing.hearingID));
-        this.router.navigate(['/', 'hearings', 'view', 'hearing-adjourned-summary', hearing.hearingID]);
-        break;
-      default:
-        this.router.navigate(['/', 'hearings', 'view']);
-        break;
+    if (!this.loadHearingRequestFromStoreRequired(hearing.exuiDisplayStatus)) {
+      this.router.navigate(['/', 'hearings', 'view']);
+    } else {
+      this.hearingStore.dispatch(new fromHearingStore.LoadHearingRequest(hearing.hearingID));
+      this.sub = this.hearingStore.select(fromHearingStore.getHearingRequestLastError).subscribe(
+        error => {
+          if (error) {
+            this.router.navigate(['/', 'hearings', 'error']);
+          } else {
+            this.router.navigateByUrl(this.getViewDetailsNavigationUrl(hearing));
+          }
+        }
+      );
```

Comment:
This seems to be less readable compare to the original code.
also the 
      this.hearingStore.dispatch(new fromHearingStore.LoadHearingRequest(hearing.hearingID)); --- this is asynch call
      this.sub = this.hearingStore.select(fromHearingStore.getHearingRequestLastError).subscribe()
--- so this subscription may not work.

I think we need to put this this.sub = this.hearingStore.select(fromHearingStore.getHearingRequestLastError).subscribe() in the ngOnInit method.

```suggestion
      case EXUIDisplayStatusEnum.CANCELLATION_REQUESTED:
        this.LoadHearingRequestAndRedirect(hearing.hearingID,  '/hearings/view/hearing-cancellation-summary'));
        break;
      case EXUIDisplayStatusEnum.CANCELLED:
        this.LoadHearingRequestAndRedirect(hearing.hearingID,  '/hearings/view/hearing-cancelled-summary'));
        break;
      case EXUIDisplayStatusEnum.COMPLETED:
                this.LoadHearingRequestAndRedirect(hearing.hearingID,  `/hearings/view/hearing-completed-summary/${hearing.hearingID}`));
       case EXUIDisplayStatusEnum.ADJOURNED:
                this.LoadHearingRequestAndRedirect(hearing.hearingID,  `/hearings/view/hearing-adjourned-summary/${hearing.hearingID}`));
   default:
        this.router.navigate(['/', 'hearings', 'view']);
        break;
        
       
   public LoadHearingRequestAndRedirect(hearingID: string, targetURL: string) {
   
         this.hearingStore.dispatch(new fromHearingStore.LoadHearingRequestAndRedirect(hearing.hearingID, targetURL: string));
```
create a new LoadHearingRequestAndRedirect action if required, the error handling can also put in the effect.

Reply:
Thank you @rayliangatos I have done the changes now.